src/main/java/org/spongepowered/common/registry/type/BlockStateRegistryModule.java

```diff
@@ -44,9 +50,47 @@ public static BlockStateRegistryModule getInstance() {
         return Holder.INSTANCE;
     }
 
+    @SuppressWarnings({"rawTypes", "unchecked"})
     @Override
     public Optional<BlockState> getById(String id) {
-        return Optional.ofNullable(this.blockStateMap.get(checkNotNull(id, "Id cannot be null!").toLowerCase(Locale.ENGLISH)));
+        final String state = checkNotNull(id, "Id cannot be null!").toLowerCase(Locale.ENGLISH);
+        final BlockState potential = this.blockStateMap.get(state);
+        if (potential != null) {
+            return Optional.of(potential);
+        }
+        if (state.contains("[")) {
```

Comment:
@Zidane I don't know if this can be improved, but how does it compare to your fuzzy maatching?